src/client.ts

```diff
@@ -134,14 +138,32 @@ export class EdcConnectorClient {
     return new FederatedCatalogController(this[innerSymbol], this.context);
   }
 
-  get addresses() {
-    return { ...this[addressesSymbol] };
+  get addresses(): Addresses {
+    return {
+      public: this.context.public,
+      control: this.context.control,
+      default: this.context.default,
+      federatedCatalogUrl: this.context.federatedCatalog,
+      identity: this.context.identity,
+      management: this.context.management,
+      presentation: this.context.presentation,
+      protocol: this.context.protocol,
+    };
+  }
+
+  addContext(
```

Comment:
maybe `setContext` will fit better, as `add` would let people think that there could be more than 1 context.

another thought: what could be the use to have the context to be modified? this could lead to unexpected side effects. 
I would keep it as an immutable client property: users could create another client with a different context or pass the context directly to the call method (IIRC all the methods permit to pass a custom context). 
wdyt?